From a17b4b5d7ffd5d5d9a5f02b350ae55075bff8d32 Mon Sep 17 00:00:00 2001
From: Mike Looijmans <milo@openpli.org>
Date: Sun, 31 Jul 2016 18:35:27 +0200
Subject: [PATCH] picload.cpp: adapt to newer giflib version

New giflib has an extra error code parameter for open/close, just
add a dummy integer pointer.
---
 lib/gdi/picload.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/gdi/picload.cpp b/lib/gdi/picload.cpp
index 580826b..61ab751 100644
--- a/lib/gdi/picload.cpp
+++ b/lib/gdi/picload.cpp
@@ -478,7 +478,7 @@ static void gif_load(Cfilepara* filepara, bool forceRGB = false)
 	int cmaps;
 	int extcode;
 
-	gft = DGifOpenFileName(filepara->file);
+	gft = DGifOpenFileName(filepara->file, &extcode);
 	if (gft == NULL)
 		return;
 	do
@@ -568,11 +568,11 @@ static void gif_load(Cfilepara* filepara, bool forceRGB = false)
 	}
 	while (rt != TERMINATE_RECORD_TYPE);
 
-	DGifCloseFile(gft);
+	DGifCloseFile(gft, &extcode);
 	return;
 ERROR_R:
 	eDebug("[ePicLoad] <Error gif>");
-	DGifCloseFile(gft);
+	DGifCloseFile(gft, &extcode);
 }
 
 //---------------------------------------------------------------------------------------------
-- 
2.1.4

